stable/redis-ha/templates/_helpers.tpl

```diff
@@ -71,3 +71,13 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{- define "redis-ha.masterGroupName" -}}
+{{- $masterGroupName := tpl ( .Values.redis.masterGroupName | default "") . -}}
+{{- $validMasterGroupName := regexMatch "^[\\w-\\.]+$" $masterGroupName -}}
```

Comment:
According to the Redis Sentinel:
```
The valid charset is A-z 0-9 and the three characters ".-_".
```
I am not sure I understand what range you are trying to match here.

Reply:
`\w Matches any letter, digit or underscore. Equivalent to [a-zA-Z0-9_].`

i added - and . to the allowed chars

and i saw that redis only accepted lowercase for master group name but i dit not check that